sdk/dapp-kit/src/components/styling/StyleMarker.tsx

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Slot } from '@radix-ui/react-slot';
+import type { ComponentPropsWithoutRef, ElementRef, ReactNode } from 'react';
+import { forwardRef } from 'react';
+
+import { styleDataAttribute } from '../../constants/styleDataAttribute.js';
+
+import './StyleMarker.css.js';
+
+type StyleMarker = {
+	children: ReactNode;
+};
+
+export const StyleMarker = forwardRef<
+	ElementRef<typeof Slot>,
+	ComponentPropsWithoutRef<typeof Slot>
+>(({ children }, forwardedRef) => (
+	<Slot ref={forwardedRef} {...styleDataAttribute}>
```

Comment:
We're going to use the `Slot` component for primitive UI elements needed for theming in a follow-up PR, so I don't have any concerns about adding this as a dependency